python/test/gluon/test_lowerings.py

```diff
@@ -238,6 +238,29 @@ def kernel(x_ptr, y_ptr, M: ttgl.constexpr, layout: ttgl.constexpr):
 ])
 
 
+# TODO: test more complex layouts with duplicated elements.
+@pytest.mark.parametrize("M, bins", [[2048, 2], [8, 512]])
+@pytest.mark.parametrize("src_layout", [ttgl.BlockedLayout([1], [THREADS_PER_WARP], [4], [0])])
+@pytest.mark.parametrize("dst_layout", [ttgl.BlockedLayout([1], [THREADS_PER_WARP], [4], [0])])
+def test_histogram(M, bins, src_layout, dst_layout, device):
+
+    @gluon.jit
+    def kernel(x_ptr, z_ptr, M: ttgl.constexpr, B: ttgl.constexpr, src_layout: ttgl.constexpr,
+               dst_layout: ttgl.constexpr):
+        offs = ttgl.arange(0, M, layout=src_layout)
+        x = ttgl.load(x_ptr + offs)
+        h = ttgl.histogram(x, B, layout=dst_layout)
+        z_offs = ttgl.arange(0, B, layout=dst_layout)
+        ttgl.store(z_ptr + z_offs, h)
+
+    torch.manual_seed(0)
+    x = torch.randint(0, bins, (M, ), dtype=torch.int32, device=device)
+    z = torch.zeros((bins, ), dtype=torch.int32, device=device)
+    z_torch = torch.histc(x.float(), bins=bins, min=0, max=bins - 1)
+    kernel[(1, )](x, z, M, bins, src_layout, dst_layout, num_warps=4)
+    assert (z == z_torch).all()
```

Comment:
NIT:
```suggestion
    torch.testing.assert_close(z, z_torch, atol=0, rtol=0)
```

Reply:
isn't assert_close meant for floating points with a tolerance.
In general `torch.testing.assert_close` is more idiomatic? I tend to use `assert (a == b).all` when I want bit width identical tensors

Reply:
It just gives a much better error messages when it fails: 
```
AssertionError: Tensor-likes are not equal!

Mismatched elements: 10 / 10 (100.0%)
Greatest absolute difference: 1.8067741394042969 at index (7,)
Greatest relative difference: 22.668630599975586 at index (6,)
```

PyTorch's test suite uses assert_close everywhere (or really a wrapped function that internally calls assert_close)